### ISSUE TITLE
"Import from OPML" does not let you select OPML file directly

### ISSUE THREAD

**Issue by wp9015362** [Etuldan/spaRSS#68]
@Etuldan :

When I was using Flym (now spaRSS) for the very first time, the following issue was confusing me a lot:

When you tap on "Import from OPML", spaRSS opens some kind of file manager, which allows you to browse through the internal storage.

What I was expecting is, that I could browse to an OPML file and select (tap on) it, to make spaRSS import it.

But contrary to my expectations, this was not how it worked.

Instead, you actually can't select any file directly and even OPML files are grayed out.

At first I thought the OPML file that I had exported to the phone previously might be incompatible with spaRSS or maybe corrupted and therefore grayed out.

But then I noticed that if you press the back button, the file manager closes and then you suddenly can select the OPML file via another dialog.

So, basically you have to select the desired folder with the file manager thing first, then you need to press the back button and then you can select the file via another dialog.

This is extremely unintuitive IMHO.

Now that I know how it works, it's kinda okay, but if there is any way this could be improved, it definitely should be improved IMHO.

In case this should be important:

The phone is a LG G2 running the stock LG Android 5.0.2 Lollipop ROM.

Regards

**Comment by Etuldan** [comment]
What file manager are you using ?
I'm not concerned with this issue with "Amaze", when I import, I have to select (in Amaze) a file, and not a directory.

**Comment by wp9015362** [comment]
> @Etuldan wrote:
> 
> What file manager are you using ?

The only file manager that I know of that I have installed is the LG stock file manager that comes preinstalled with the LG G2 stock Android 5.0.2 Lollipop ROM.

But I am fairly sure that spaRSS does __not__ open the the LG stock file manager that comes preinstalled with the LG G2 stock Android 5.0.2 Lollipop ROM.

It seems to be something else.

It might also be worth mentioning, that it is not really a file manager, it doesn't allow you to modify anything. It only allows you to browse through the file system, nothing else.

Also, when this file browsing thing is being opened by spaRSS and when I have a look at the running apps, then I only see spaRSS. Not any other additional file manager app or something like that.

It kinda looks like as if this file browsing thing would be built into spaRSS.

I don't know how to open this file browsing thing other than with spaRSS import from OPML option.

When I tap on "Import from OPML", this file browsing thing opens and offers the following options:

Open from:
- Recent
- Downloads
- Internal storage

Maybe it's some kind of basic core feature of Android which is being opened by spaRSS when spaRSS doesn't find a compatible file manager or something like that?

PS:

Okay, after having written this, I have now installed Amaze File Manager.

Now when I tap on "Import from OPML", spaRSS still opens this basic file browsing thing that I was talking about above.

But now this basic file browsing thing offers the following options:

Open from:
- Recent
- Downloads
- Internal storage
- Amaze

When tapping on "Amaze" it opens Amaze File Manager.

When opening Amaze File Manager as described above, it's now also possible to directly select an OPML file via Amaze File Manager.

It does not offer to open the LG stock file manager though. I have no idea why.

Strange.

Anyway, Amaze File Manager seems to be quite nice and it's even open source, so I guess I'll rather use Amaze File Manager from now on instead of the LG stock file manager :+1: :blush: :relaxed: :stuck_out_tongue_winking_eye: :wink:.

Regards

**Comment by Etuldan** [comment]
I installed an other file manager, and it's displayed in spaRSS (while none of them is hardcoded in spaRSS files)
So I think that the "LG file manager" didn't broadcast its status of "File Manager" to the other apps

**Comment by notlily** [comment]
OPML import seems to be broken entirely, whether using the Android file picker (greyed out as above), Cyanogenmod's file manager, or Amaze. "The selected file could not be imported". 

I'd guess it was the file that's the problem, but it imports into Flym just fine. Flym's OPML output doesn't work either.

**Comment by Etuldan** [comment]
Could you upload the OPML file (even if it's not related to this one, as it works on Flym) ?

**Comment by Etuldan** [comment]
I cannot reproduce on my 'debug' build.
I'll try this evening with the stable one.

Also, it might be related to the feed "moetron" (as it's title has some non-ascii characters)
Could you try on your device after removing the line in your OPML file ?

**Comment by smith153** [comment]
This is probably still an issue. I've spent the last hour trying to import from my old phone. I finally managed to get an export to upload into Drive, but once I downloaded it to local storage on the new phone, the selection for the file is grayed out.

**Comment by smith153** [comment]
Finally managed to get it. For some reason when I would download it with Drive, the only options I had in the (samsung) File Manager were to delete or rename. Within Drive I set the file to be "Available Offline" and redownloaded it. Then I had the option to be able to move the file, with the samsung file manager. Using the import feature in sparss still showed all files as grayed out though. So I copied the file to the root of the storage directory and then when I would back out of  the import sparss interface, it would prompt to use the file at the root of the directory, which would then work fine.

**Comment by Dunedan** [comment]
Same issue still present under LineageOS 14.1 and its integrated file manager. Thanks @wp9015362 for the tip with the back button.

**Comment by clessley** [comment]
I too can confirm smith153's technique (only way) works LineageOS using stock file manager:

> copied the [opml] file to the root of the [internal] storage directory and then when I would back out of the import sparss interface, it would prompt to use the file at the root of the directory, which would then work fine.

**Comment by bjoernv** [comment]
If I select the menu item "Importiere von OPML" ("Import from OPML") I get a message dialog without any buttons, but the following text "From /storage/emulated/0/Android/data/de.bernd.shands...".
(tested on Moto G5 Plus, Android 7.0)

**Comment by kousu** [comment]
This is still an issue on LineageOS with the stock file manager, which makes me think it's probably an issue on any newish ROM too.

All files are greyed out in the file manager, and only folders are selectable. Closing out of the file manager than prompts for all files. It makes it seem like spaRSS is asking for a *folder* to import from and then generating its own file picker, rather than using the built in file picker.

**Comment by pbryan** [comment]
FYI, I just ran into this problem myself on LineageOS. Happy to provide any information on request.
